src/cascade/model/grids.py

```diff
@@ -0,0 +1,174 @@
+            lower_time = float("-inf")
+        else:
+            lower_time = self._time_slice.start
+
+        if self._time_slice.stop is None:
+            upper_time = float("inf")
+        else:
+            upper_time = self._time_slice.stop
+
+        return (lower_age, upper_age, lower_time, upper_time)
+
+    @property
+    def prior(self):
+        lower_age, upper_age, lower_time, upper_time = self._rectangle()
+        if lower_age != upper_age or lower_time != upper_time:
+            raise NotImplementedError("Currently only point queries are supported")
+
+        return self._parent._prior_at_point(lower_age, lower_time)
+
+    @prior.setter
+    def prior(self, p):
+        self._parent._push_prior(*self._rectangle(), p)
+
+
+class PriorGrid:
+    """Represents priors for rectangular regions of an underlying age-time grid.
+
+    Args:
+        grid: The underlying age-time grid
+        hyper_prior: The prior for the priors in this grid
+
+    Examples:
+        >>> grid = AgeTimeGrid.uniform(age_start=0,age_end=120,age_step=5,time_start=1990,time_end=2018,time_step=1)
+        >>> d_time = PriorGrid(grid)
+        >>> #Set a prior for the whole grid:
+        >>> d_time[:, :].prior = GaussianPrior(0, 0.1)
+        >>> #Set a prior for a band of ages
+        >>> d_age[0:15, :].prior = GaussianPrior(1, 0.01)
+        >>> #Or a single year
+        >>> d_time[:, 1995].prior = GaussianPrior(0, 3)
+    """
+
+    def __init__(self, grid, hyper_prior=None):
+        self._grid = grid
+        self._priors = list()
+        self.hyper_prior = hyper_prior
+
+    def __getitem__(self, slices):
+        try:
+            if len(slices) != 2:
+                raise ValueError("Region must be specified in both age and time")
+        except TypeError:
+            raise ValueError("Region must be specified in both age and time")
+
+        age_slice, time_slice = slices
+        if not isinstance(age_slice, slice):
+            age_slice = slice(age_slice, age_slice)
+        if not isinstance(time_slice, slice):
+            time_slice = slice(time_slice, time_slice)
+        if age_slice.step is not None or time_slice.step is not None:
+            raise ValueError("Step size must not be specified. It is defined in the underlying grid")
+
+        return _RegionView(self, age_slice, time_slice)
+
+    def _push_prior(self, lower_age, upper_age, lower_time, upper_time, prior):
+        """Push a new region and prior combination onto the stack. When
+        querying, more recent priors take precedence.
+        """
+        _validate_region(self._grid, lower_age, upper_age, lower_time, upper_time)
+        self._priors.append(((lower_age, upper_age, lower_time, upper_time), prior))
+
+    def _prior_at_point(self, age, time):
+        """Find the prior for a particular point on the age-time grid.
+        """
+        final_prior = None
+        for ((lower_age, upper_age, lower_time, upper_time), prior) in self._priors:
```

Comment:
This is similar to reversing the _priors and returning at the first one found.

Reply:
So it is. That's certainly a simpler implementation. I'm going to switch to that.